src/Microsoft.IdentityModel.Logging/IdentityModelTelemetryUtil.cs

```diff
@@ -0,0 +1,113 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Collections.Concurrent;
+using System.Reflection;
+using System.Net.Http;
+
+namespace Microsoft.IdentityModel.Logging
+{
+    /// <summary>
+    /// Provides a way to add and remove telemetry data.
+    /// </summary>
+    public static class IdentityModelTelemetryUtil
+    {
+        internal static readonly ConcurrentDictionary<string, string> telemetryData = new ConcurrentDictionary<string, string>();
+        internal const string skuTelemetry = "x-client-SKU";
+        internal const string versionTelemetry = "x-client-Ver";
+        internal const string osTelemetry = "x-client-OS";
+        internal static List<string> defaultTelemetryValues = new List<string> { skuTelemetry, versionTelemetry };
+
+        /// <summary>
+        /// Get the string that represents the client SKU.
+        /// </summary>
+        public static string ClientSku { get;} =
+#if NET45
+            "ID_NET45";
+#elif NET461
+            "ID_NET461";
+#elif NET472
+            "ID_NET472";
+#elif NETSTANDARD2_0
+            "ID_NETSTANDARD2_0";
+#endif
+
+        /// <summary>
+        /// Get the string that represents the client version.
+        /// </summary>
+        public static string ClientVer { get; } = typeof(IdentityModelTelemetryUtil).GetTypeInfo().Assembly.GetName().Version.ToString();
+
+        static IdentityModelTelemetryUtil()
+        {
+            telemetryData[skuTelemetry] = ClientSku;
+            telemetryData[versionTelemetry] = ClientVer;
+        }
+
+        /// <summary>
+        /// Adds a key and its value to the collection of telemetry data.
+        /// </summary>
+        /// <param name="key"> Represents the name of the telemetry.</param>
+        /// <param name="value"> Represents the value of the telemetry.</param>
+        public static void AddTelemetryData(string key, string value)
+        {
+            if (string.IsNullOrEmpty(key))
+                throw LogHelper.LogArgumentNullException(nameof(key));
+
+            CheckIfDefaultTelemetry(key);
+            if (value == null)
+                RemoveTelemetryData(key);
+            else
+                telemetryData[key] = value;
```

Comment:
Is there a reason why we need to call Remove() and can't just override the value by using the setter (as in the 'else' case)?

Reply:
That's from my assumption that we do not want to allow keys with null values. I am open to update if we think its ok to have them.